vault/resource_database_secret_backend_connection.go

```diff
@@ -1510,8 +1511,14 @@ func setDatabaseConnectionDataWithUserPass(d *schema.ResourceData, prefix string
 
 	data["username"] = d.Get(prefix + "username")
 
-	if v, ok := d.GetOk(prefix + "password"); ok {
-		data["password"] = v.(string)
+	// Vault does not return the password in the API. If the root credentials have been rotated, sending
+	// the old password in the update request would break the connection config. Thus we only send it,
+	// if it actually changed to still support updating it for non-rotated cases.
+	passwordKey := prefix + consts.FieldPassword
+	if v, ok := d.GetOk(passwordKey); ok {
+		if d.IsNewResource() || d.HasChange(passwordKey) {
```

Comment:
The first `d.IsNewResource` case might be redundant. Will double check and remove if it is so